connectors/services/job_scheduler.py

```diff
@@ -0,0 +1,206 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+"""
+Event loop
+
+- polls for work by calling Elasticsearch on a regular basis
+- instantiates connector plugins
+- mirrors an Elasticsearch index with a collection of documents
+"""
+import asyncio
+import os
+import time
+
+from connectors.byoc import (
+    SYNC_DISABLED,
+    BYOIndex,
+    ConnectorUpdateError,
+    DataSourceError,
+    JobTriggerMethod,
+    ServiceTypeNotConfiguredError,
+    ServiceTypeNotSupportedError,
+    Status,
+    SyncJobIndex,
+    e2str,
+)
+from connectors.logger import logger
+from connectors.services.base import BaseService
+from connectors.filtering.validation import (
+    FilteringValidationState,
+    InvalidFilteringError,
+)
+
+
+class JobSchedulerService(BaseService):
+    def __init__(self, config):
+        super().__init__(config)
+        self.errors = [0, time.time()]
+        self.service_config = self.config["service"]
+        self.idling = self.service_config["idling"]
+        self.hb = self.service_config["heartbeat"]
+        self.connector_index = None
+        self.sync_job_index = None
+
+    def raise_if_spurious(self, exception):
```

Comment:
ah! that's what I was saying earlier. Move it to `BaseService`

Reply:
I have done it in this PR: https://github.com/elastic/connectors-python/pull/337